stablehlo/compatibility/dialect/VersionedStablehloBase.td

```diff
@@ -0,0 +1,182 @@
+/* Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+   Copyright 2022 The VersionedHLO Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#ifndef STABLEHLO_COMPATIBILITY_VERSIONEDHLO_BASE
+#define STABLEHLO_COMPATIBILITY_VERSIONEDHLO_BASE
+
+include "mlir/IR/OpBase.td"
+
+//===----------------------------------------------------------------------===//
+// Versioned Dialect Constants and Declaration
+//===----------------------------------------------------------------------===//
+
+def VersionedHLO_Dialect : Dialect {
```

Comment:
Can fix that. Moved because the dependency structure seemed strange, and I encountered build errors when I was getting started on this work. `*Enums.td` and `*Attrs.td` requires the definition of the Dialect, and are included by `*Ops.td` where the Dialect is defined. Base felt like a fitting place to put the definitions that all other `Vhlo*.td` files needed.

StableHLO and CHLO get around this by including these two files after a few lines of declarations in the tablegen, like a `.inc` include, which LLVM coding style would denote as "[hidden dependencies](https://llvm.org/docs/CodingStandards.html#include-style)" (for C++ at least, maybe this is common practice for tablegen).